src/types/Hashed.ts

```diff
@@ -0,0 +1,29 @@
+import * as bcryptC from 'bcrypt'
+import * as Promise from 'bluebird'
+import * as _ from 'lodash'
+
+const bcrypt:any = Promise.promisifyAll(bcryptC)
```

Comment:
It might also be worth updating to the latest bcrypt/bcryptjs versions as I think they can return a Promise directly